generators/Exercises/Generators/PrimeFactors.cs

```diff
@@ -1,10 +1,18 @@
-﻿using Exercism.CSharp.Output;
+﻿using System.Linq;
+using Exercism.CSharp.Output;
 
 namespace Exercism.CSharp.Exercises.Generators
 {
     public class PrimeFactors : GeneratorExercise
     {
-        protected override void UpdateTestMethod(TestMethod testMethod) =>
+        protected override void UpdateTestMethod(TestMethod testMethod)
+        {
             testMethod.Input["value"] = (long)testMethod.Input["value"];
+
+            if (testMethod.Expected is int[])
+            {
+                testMethod.Expected = ((int[])testMethod.Expected).Select(l => (long)l).ToArray();
```

Comment:
I'm not entirely sure if it works, but you could check to see if instead of the select and the lambda, you could use "Cast<long>".

Reply:
I tried the following:
`testMethod.Expected = expected.Cast<long>();`

But that gives the following error:
`Unable to cast object of type 'System.Int32' to type 'System.Int64'.'`

I'd expect that error in the other sense, but not here... 🤔